privval/utils.go

```diff
@@ -15,10 +14,11 @@ import (
 // report that a connection timeout occurred. This detects both fundamental
 // network timeouts, as well as ErrConnTimeout errors.
 func IsConnTimeout(err error) bool {
-	switch errors.Cause(err).(type) {
-	case EndpointTimeoutError:
+	_, ok := errors.Unwrap(err).(timeoutError)
```

Comment:
```suggestion
	 ok := errors.Is(err, timeoutError)
```
?

Reply:
nope. Is requires pointer to a struct. timeoutError is an interface